### PR TITLE
Fix build errors when TRACK_SYNC is defined

### PR DESCRIPTION
**Pull request by parjong**
This commit fixes build errors related with TRACK_SYNC feature (as the first step to #13120) .

### CODE REVIEW

File: src/vm/syncblk.inl [hunk 1]
```diff
@@ -178,7 +178,7 @@ FORCEINLINE AwareLock::LeaveHelperAction AwareLock::LeaveHelper(Thread* pCurThre
     _ASSERTE((size_t)m_MonitorHeld & 1);
     _ASSERTE(m_Recursion >= 1);
 
-#if defined(_DEBUG) && defined(TRACK_SYNC)
+#if defined(_DEBUG) && defined(TRACK_SYNC) && !defined(CROSSGEN_COMPILE)
```
Comment (janvorli) [inline src/vm/syncblk.inl]: I think that if you defined TRACK_SYNC in both clrdefinitions.cmake and crossgen.cmake, you would not need this added condition.

Comment (parjong) [inline src/vm/syncblk.inl]: @janvorli ``Thread`` does not have ``m_pTrackSync`` when ``CROSSGEN_COMPILE`` is set (please refer to the definition of ``Thread`` at the beginning of ``threads.h`` for details).

So, setting ``TRACK_SYNC`` in crossgen.cmake can't address this issue. Do we need to introduce ``m_pTrackSync`` in ``Thread`` for crossgen?

Comment (janvorli) [inline src/vm/syncblk.inl]: Oh, ok, I've missed that. Then it makes sense the way you did it.

File: src/vm/threads.h [hunk 2]
```diff
@@ -515,6 +515,9 @@ typedef Thread::ForbidSuspendThreadHolder ForbidSuspendThreadHolder;
 // Each thread has a stack that tracks all enter and leave requests
 struct Dbg_TrackSync
 {
+    Dbg_TrackSync() = default;
```
Comment (janvorli) [inline src/vm/threads.h]: Why is this needed? The default constructor is always present by default.

Comment (parjong) [inline src/vm/threads.h]: I added it just for some consistency. I'll remove it.

**Comment by parjong** [comment]
@dotnet-bot test Tizen armel Cross Debug Build please
